corehq/apps/hqadmin/service_checks.py

```diff
@@ -0,0 +1,170 @@
+"""
+A collection of functions which test the most basic operations of various services.
+"""
+from collections import namedtuple
+import datetime
+import json
+import logging
+from StringIO import StringIO
+import time
+
+from django.core import cache
+from django.conf import settings
+from django.contrib.auth.models import User
+from restkit import Resource
+import requests
+from soil import heartbeat
+
+from corehq.apps.app_manager.models import Application
+from corehq.apps.change_feed.connection import get_kafka_client_or_none
+from corehq.apps.es import GroupES
+from corehq.blobs import get_blob_db
+from corehq.elastic import send_to_elasticsearch
+from corehq.util.decorators import change_log_level
+
+ServiceStatus = namedtuple("ServiceStatus", "success msg")
+
+
+def check_redis():
+    if 'redis' in settings.CACHES:
+        import redis
+        rc = cache.caches['redis']
+        redis_api = redis.StrictRedis.from_url('%s' % rc._server)
+        memory = redis_api.info()['used_memory_human']
+        result = rc.set('serverup_check_key', 'test')
```

Comment:
very minor: might want to add a timeout to this so the key/value pair is automatically cleaned up